### PR TITLE
Request context doc and example

### PR DESCRIPTION
**Pull request by gsserge**
* Add request context section under `Data sharing` in `lib.rs`.
* Add request duration example.
* Update links and tidy up docs and readme a bit.

I've taken a liberty to trim down readme and lib.rs by removing the second paragraph and benchmarks in order for the potential users to get to the examples faster. I understand that this might be completely subjective and have no problem with dropping the changes if you like.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,11 +38,13 @@ Add this to your `Cargo.toml` file:
...
 [dependencies]
 routerify = "2.0.0-beta-4"
+hyper = "0.14"
+tokio = { version = "1", features = ["full"] }
 ```
 
 ## Basic Example
 
-A simple example using `Routerify` with [hyper.rs](https://hyper.rs/) would look like the following:
+A simple example using `Routerify` with `hyper` would look like the following:
```
Comment (seanpianka) [inline README.md:47]:
```suggestion
A simple example using `Routerify` with [`hyper`](https://hyper.rs/) would look like the following:
```

The best of both worlds?

**Comment by rousan** [comment]
Everything looks fine. But, we could show the benchmarks data in README and lib.rs to showcase routerify performance as quick comparison between existing libraries.

**Comment by gsserge** [comment]
> Everything looks fine. But, we could show the benchmarks data in README and lib.rs to showcase routerify performance as quick comparison between existing libraries.

Of course. I'm going to update `routerify-benchmark` to include the latest version of each of the framework.

**Review by rousan — APPROVED** [review]

**Comment by gsserge** [comment]
Benchmarks update https://github.com/routerify/routerify-benchmark/pull/3

**Comment by seanpianka** [comment]
Thoughts on changing the copy for the features?

Routerify's core features:

- 🌀 Allows defining complex routing using **scopes** and **middlewares**
- 🚀 Fast route matching using [`RegexSet`](https://docs.rs/regex/1.4.3/regex/struct.RegexSet.html)
- 🍺 Route handlers return any type implementing [HttpBody](https://docs.rs/hyper/0.14.4/hyper/body/trait.HttpBody.html)
- ❗ Flexible error handling strategy
- 💁 [`WebSocket` support](https://github.com/routerify/routerify-websocket) out of the box.
- 🔥 Route and middleware handlers may share state
- 🍗 Extensive documentation and [examples](https://github.com/routerify/routerify/tree/master/examples)

**Comment by gsserge** [comment]
Looks good! Can you open another PR changing this in both readme and lib.rs?

As a non-native English speaker I'm still puzzled: what is the plural form for middleware? :) Is it middleware or middleware**s**?

**Comment by seanpianka** [comment]
I typically use "middleware" as both a singular and plural form. [There are instances where you would use "middlewares", though](https://english.stackexchange.com/questions/257120/middleware-vs-middlewares). 

It's the difference between ["countable" and "uncountable" nouns](https://dictionary.cambridge.org/grammar/british-grammar/nouns-countable-and-uncountable)... and in this scenario, where we reference any (uncountable) number of potential middlewares (🙂), using "middleware" feels appropriate.

**Pull request merged**